rs/crypto/utils/canister_threshold_sig/src/lib.rs

```diff
@@ -22,3 +25,26 @@ pub fn derive_threshold_public_key(
         }
     })
 }
+
+/// Derives the vetKD public key from the specified `master_public_key` for
+/// the given `extended_derivation_path`.
+pub fn derive_vetkd_public_key(
+    master_public_key: &MasterPublicKey,
+    extended_derivation_path: &ExtendedDerivationPath,
+) -> Result<Vec<u8>, VetKdPublicKeyDeriveError> {
```

Comment:
First I thought that we could just extend `fn derive_threshold_public_key` and decide based on the `MasterPublicKey::algorithm_id` which derivation to do (vetkd, others), but the return value of that function is a `PublicKey` which has a (non-optional) `chain_key: Vec<u8>`, so this is not ideal as we'd have to return an empty `chain_key`. So, having a separate `derive_vetkd_public_key` seems a good way forward.